sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -392,6 +392,156 @@ def get_access_control(self, upn=None,  # type: Optional[bool]
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @staticmethod
+    def _set_access_control_recursive_options(mode, acl=None, **kwargs):
+        # type: (Optional[ContentSettings], Optional[Dict[str, str]], **Any) -> Dict[str, Any]
+
+        options = {
+            'mode': mode,
+            'timeout': kwargs.pop('timeout', None),
+            'continuation': kwargs.pop('continuation', None),
+            'max_records': kwargs.pop('batch_size', None),
+            'acl': acl,
+            'cls': return_headers_and_deserialized}
+        options.update(kwargs)
+        return options
+
+    def set_access_control_recursive(self,
+                                     acl=None,  # type: Optional[str]
+                                     progress_callback=None,
+                                     **kwargs):
+        # type: (...) -> Dict[str, Union[str, datetime]]
+        """
+        Sets the Access Control on a path and sub-paths.
+
+        :param acl:
+            Sets POSIX access control rights on files and directories.
+            The value is a comma-separated list of access control entries. Each
+            access control entry (ACE) consists of a scope, a type, a user or
+            group identifier, and permissions in the format
+            "[scope:][type]:[id]:[permissions]".
+        :type acl: str
+        :param progress_callback:
+            Callback where the caller can track progress of the operation
+            as well as collect paths that failed to change Access Control.
+        :type progress_callback: func(~azure.storage.filedatalake._generated.models.SetAccessControlRecursiveResponse)
+        :keyword str continuation:
+            Optional continuation token that can be used to resume previously stopped operation.
```

Comment:
When things involve continuation we tend to use other patterns... For example FileShares.close_handles, or pagable/pollable operations. What is the use case for continuation here?

Reply:
I didn't go with pageable as the user is not really consuming any information (like the get_paths method), the set acl operations should be executed continuously as the previous call succeeds.

This is similar to FileShares.close_handles, but with extra controls (how many files at a time, how many batches in total) and more information given back (updates from each individual call). The main reason is that this can be a really lengthy operation, and can fail at any time, so the cx needs to record the progress as we go, so that they can assess the situation and resume if needed. FileShares.close_handles should be short-running in general.

Reply:
Can we name the parameter `continuation_token`? This will be consistent with the naming convention used in paging.

Reply:
Your comment that this can be "a really lengthy operation" leads to me to think we should use our polling APIs for this. That would give users a consistent experience for handling this, monitoring state etc.

Reply:
@annatisch @zezha-msft We had very similar discussion for .NET. I looped you in on email thread.